CONTRIBUTING.md

```diff
@@ -0,0 +1,374 @@
+
+### Feature request
+
+Feature requests are welcome, but take a moment to find out whether your idea fits with the scope and aims of the project.
+It's up to you to make a strong case to convince the project's developers of the merits of this feature.
+Please provide as much detail and context as possible and fill in the information requested in the [feature request template](https://github.com/janus-idp/backstage-plugins/issues/new?labels=kind%2Fenhancement%2Cstatus%2Ftriage&projects=janus-idp/2&template=enhancement.md).
+
+### New plugin request
+
+Plugins are a great way to extend Backstage capabilities and are an integral part of this repository.
+Please provide as much detail and context as possible and fill the information requested in the [plugin suggestion template](https://github.com/janus-idp/backstage-plugins/issues/new?labels=plugin%2Cstatus%2Ftriage&projects=janus-idp/2&template=plugin.yaml&title=%F0%9F%94%8C+Plugin%3A+).
+
+## Submitting a Pull Request
+
+Good pull requests, whether patches, improvements, or new features, are a fantastic help.
+They should remain focused in scope and avoid containing unrelated commits.
+
+**Please ask first** before embarking on any significant pull requests (e.g. implementing features, refactoring code), otherwise you risk spending a lot of time working on something that the project's maintainers might not want to merge into the project.
+
+If you have never created a pull request before, welcome 🎉 😄.
+[Here is a great tutorial](https://opensource.guide/how-to-contribute/#opening-a-pull-request) on how to send one 🙂
+
+Here is a summary of the steps to follow:
+
+1. [Set up the workspace](#set-up-the-workspace)
+2. If you cloned a while ago, get the latest changes from upstream and update dependencies:
+
+```bash
+$ git checkout main
+$ git pull upstream main
+$ yarn install
+```
+
+3. Create a new topic branch (off the main project development branch) to contain your feature, change, or fix:
+
+```bash
+$ git checkout -b <topic-branch-name>
+```
+
+4. Make your code changes, following the [Coding rules](#coding-rules)
+5. Push your topic branch up to your fork:
+
+```bash
+$ git push origin <topic-branch-name>
+```
+
+6. [Open a Pull Request](https://help.github.com/articles/creating-a-pull-request/#creating-the-pull-request) with a clear title and description.
+
+**Tips**:
+
+- For ambitious tasks, open a Pull Request as soon as possible with the `[WIP]` prefix in the title, in order to get feedback and help from the community.
+- [Allow Janus-IDP maintainers to make changes to your Pull Request branch](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
+  This way, we can rebase it and make some minor changes if necessary.
+  All changes we make will be done in the new commit, and we'll ask for your approval before merging them.
+
+## Coding rules
```

Comment:
Are there any naming rules that should be added here? Like file, plugin or folder names?

Reply:
Good call, I'll add that. 👍